windows/inc/leatherman/windows/user.hpp

```diff
@@ -0,0 +1,30 @@
+/**
+ * @file
+ * Declares utility functions for querying user properties
+ */
+#pragma once
+
+#include <string>
+
+namespace leatherman { namespace windows { namespace user {
+
+    /**
+     * Determines whether the current process has Administrator privileges and can be expected to succeed at
+     * tasks restricted to Administrators.
+     * @return True if the current process has Administrator privileges, otherwise false.
+     */
+    bool is_admin();
+
+    /**
+     * Query token membership to determine whether the current user is a member of the Administrators group.
+     * @return True if user is an Administrator, otherwise false.
+     */
+    bool check_token_membership();
```

Comment:
These were extracted from https://github.com/puppetlabs/puppet/blob/003d2b73b4730953092bb7e8b7bc723d2210c999/lib/puppet/util/windows/user.rb, and will likely be needed in a future native client. Both are marked module functions, so I suspect they would be useful elsewhere.